cli/tools/fmt.rs

```diff
@@ -1584,40 +1578,36 @@ fn get_resolved_yaml_config(
   }
 }
 
-struct FileContents {
-  text: String,
-  had_bom: bool,
-}
-
-fn read_file_contents(file_path: &Path) -> Result<FileContents, AnyError> {
+fn read_file_contents(file_path: &Path) -> Result<String, AnyError> {
   let file_bytes = fs::read(file_path)
     .with_context(|| format!("Error reading {}", file_path.display()))?;
-  let had_bom = file_bytes.starts_with(&[0xEF, 0xBB, 0xBF]);
-  // will have the BOM stripped
+  let bom = &[0xEF, 0xBB, 0xBF];
+
+  // BOM stripped
+  let file_bytes = if file_bytes.starts_with(bom) {
+    file_bytes[bom.len()..].to_vec()
+  } else {
+    file_bytes
+  };
+
   let charset =
     deno_media_type::encoding::detect_charset_local_file(&file_bytes);
-  let text =
-    deno_media_type::encoding::decode_owned_source(charset, file_bytes)
-      .with_context(|| {
-        anyhow!("{} is not a valid UTF-8 file", file_path.display())
-      })?;
+  let text = deno_media_type::encoding::decode_owned_source(
+    charset,
+    file_bytes.to_vec(),
+  )
+  .with_context(|| {
+    anyhow!("{} is not a valid UTF-8 file", file_path.display())
+  })?;
 
-  Ok(FileContents { text, had_bom })
+  Ok(text)
```

Comment:
I think we might need to still track when the file had a bom so that we can always overwrite that file to not have a bom.